nixos/modules/services/mail/rss2email.nix

```diff
@@ -0,0 +1,226 @@
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.rss2email;
+        defaultText = "pkgs.rss2email";
+        description = "The rss2email package to use.";
+      };
+
+      to = mkOption {
+        type = types.str;
+        description = "Mail address to which to send emails";
+      };
+
+      interval = mkOption {
+        type = types.str;
+        default = "1h";
+        description = "How often to check the feeds, in systemd interval format";
+      };
+
+      configuration = mkOption {
+        type = types.attrs;
+        description = ''
+          The configuration to give rss2email.
+
+          Default will use system-wide <literal>sendmail</literal> to send the
+          email. This is rss2email's default when running
+          <literal>r2e new</literal>.
+
+          This set contains key-value associations that will be set in the
+          <literal>[DEFAULT]</literal> block along with the
+          <literal>to</literal> parameter.
+        '';
+        example = lib.literalExample ''
+          {
+            from = "user@rss2email.invalid";
+
+            # You likely want to use something like this for css
+            css = builtins.replaceStrings ["\n"] [""] '''
+              body {
+                font: 15pt Arial;
+              }
+            ''';
+          }
+        '';
+        default = default-config;
+      };
+
+      feeds = mkOption {
+        description = "The feeds to watch.";
+        type = types.attrsOf (types.submodule {
+          options = {
+            url = mkOption {
+              type = types.str;
+              description = "The URL at which to fetch the feed.";
+            };
+
+            to = mkOption {
+              type = with types; nullOr str;
+              default = null;
+              description = ''
+                Email address to which to send feed items. Defaults to
+                <literal>rss2email.to</literal>.
+              '';
+            };
+          };
+        });
+      };
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    users.groups = {
+      rss2email.gid = config.ids.gids.rss2email;
+    };
+
+    users.users = {
+      rss2email = {
+        description = "rss2email user";
+        uid = config.ids.uids.rss2email;
+        group = "rss2email";
+      };
+    };
+
+    systemd.services.rss2email = {
+      preStart = ''
+        mkdir -p /var/rss2email
+        chmod 700 /var/rss2email
+
+        cp ${conf} /var/rss2email/conf.cfg
+        if [ ! -f /var/rss2email/db.json ]; then
+          echo '{"version":2,"feeds":[]}' > /var/rss2email/db.json
+        fi
+
+        chown -R rss2email:rss2email /var/rss2email
+      '';
+      path = [ pkgs.system-sendmail ];
+      serviceConfig = {
+        ExecStart =
+          "${cfg.package}/bin/r2e -c /var/rss2email/conf.cfg -d /var/rss2email/db.json run";
+        User = "rss2email";
+        PermissionsStartOnly = "true";
```

Comment:
I think this would be a good usecase for the `RuntimeDirectory` and `DynamicUser` systemd options. Take a look at them (`man 5 systemd.exec`). And if the `db.json` file is state, you can probably use `StateDirectory`. Would simplify this service a lot.

Reply:
I thought of `DynamicUser` before rejecting it due to `db.json`… at the time I didn't know about `StateDirectory`, now I do, thanks! :)